gitdiff/file_header.go

```diff
@@ -270,6 +276,11 @@ func parseGitHeaderData(f *File, line, defaultName string) (end bool, err error)
 		parse  func(*File, string, string) error
 	}{
 		{"@@ -", true, nil},
+		// TODO '@@@ -' is the simplest form of the combined diff; it
+		//   could be any number of '@' - one per parent + 1 '@'.
+		//   The fix involves adding a separate '@' count discovery
+		//   before this block (as '@' is the most commonly encountered structure).
```

Comment:
Adding this fix would add even more complexity to this PR.  Rather than try to tackle that here, I'm proposing to only handle the simple case first, and add in the more-than-2 scenario later.